First swipe to get tinyMCEs in #multiple fields

bendoh opened the pull request:
## Summary

Addresses issue #38 – Allow for `#multiple` fields to contain `mce`-type fields, which involves some JavaScript hoop-jumping-through.
## Relavant Ticket(s)

Issue #38

Commit:
First swipe to get tinyMCEs in #multiple fields

balbuf commented on inc/class-wp-forms-api.php:
```diff
@@ -925,6 +927,29 @@ static function process_element( $element, &$values, &$input ) {
+				'textarea_name' => $id,
+				'textarea_rows' => count( $lines ),
+			)  );
+			$results[$id] = ob_get_clean();
```
I have no idea about the performance implications but do you think it would be better to call `ob_start()` once outside of the loop and then `ob_get_contents()` and `ob_clean()` in the loop?

bendoh replied:
I did a little research and couldn't find anything about the performance impact of starting and ending output buffering multiple times – the only difference that i see is that the output is potentially contained in memory slightly longer, but I think that happens with the PHP-level output buffering that is on by default anyway, and so it would just be a merely syntactical change that i think would be more confusing than having the start/get/end piece within the loop.

balbuf commented on inc/class-wp-forms-api.php:
```diff
@@ -925,6 +927,29 @@ static function process_element( $element, &$values, &$input ) {
+
+		foreach ( $bodies as $id => $body ) {
+			ob_start();
+			$lines = preg_split( '/$/m', $body );
```
If you are only concerned with the count, would it be better to use `substr_count()` or even `preg_match_all()`?

bendoh replied:
So `substr_count()` wouldn't deal with differing line endings. If for some reason a CRLF line ending was entered, most likely from a windows machine – I'd have to be able to accept CR, LF or CRLF, which using `/$/m` allows for.

balbuf replied:
Yeah but you don't actually need to break on that, you just need to count them - isn't it always going to be CRLF or just LF? Do some systems actually just do CR? In any case, you could switch it to `preg_match_all()` since the return value is number of matches (may need to add 1 tho)

balbuf commented on inc/wp-forms-api.js:
```diff
@@ -13,13 +13,49 @@
+	function loadMCETemplates(content, editor) {
+		wp.ajax.post('wp_form_get_mce_templates', {
+			content: content,
+			type: getUserSetting('editor')
```
It doesn't look like `type` is being used in the ajax handler?

bendoh replied:
Yeah this isn't complete yet, I was just getting the base code in there – probably not ready for a full review yet~!

balbuf commented on inc/wp-forms-api.js:
```diff
@@ -13,13 +13,49 @@
+				var config = _.clone(tinyMCEPreInit.mceInit.content);
+
+				config.selector = id;
+				config.toolbar1 = ['bold', 'italic', 'underline', 'blockquote', 'strikethrough', 'bullist', 'numlist', 'alignleft', 'aligncenter', 'alignright', 'undo', 'redo', 'link', 'unlink'].join(',');
```
Shouldn't these come from WP somehow? Since normally the toolbar can be filtered I believe...

balbuf commented:
so far so good!

Commit:
Additional templating work, add stub test.

Need to put together a better test scenario for this (and the whole
plugin in general!)

balbuf commented:
_nudge, nudge_ 🐤

balbuf commented:
BUMP! I need this for CRE plzzzzz @bendoh

bendoh commented:
I know but it's not easy and I'm not exactly swimming in free time here!

balbuf commented:
Kk that's fine - I'm just going to add a couple extra MCE fields that they can use instead of letting them have an arbitrary number.

jcowher commented:
FYI... I did this exact thing back in my WPMU Dev days. Want me to see if I can dig that code up?
